webserver.cc

```diff
@@ -66,7 +67,28 @@ Server::Server(configArguments configArgs, std::map<std::string, std::vector<std
 
 void Server::doAccept()
 {
-    std::shared_ptr<session> sesh = std::make_shared<session>(io_service, configContent.handlerMapping, configContent.defaultHandler, configContent.logFileName);
+    NginxConfigParser config_parser;
+    NginxConfig config_out;
+    std::shared_ptr<session> sesh;
+    if (!config_parser.Parse(configFileName.c_str(), &config_out))
```

Comment:
Maybe you can add a comment here explaining why you parse the config here as well.